src/lib/x509/x509_ext.h

```diff
@@ -500,6 +500,66 @@ class OCSP_NoCheck final : public Certificate_Extension {
       void decode_inner(const std::vector<uint8_t>&) override;
 };
 
+/**
+* TNAuthList extension
+*
+* RFC8226 Secure Telephone Identity Credentials
+*   https://www.rfc-editor.org/rfc/rfc8226#section-9
+*/
+class BOTAN_PUBLIC_API(3, 6) TNAuthList final : public Certificate_Extension {
+   public:
+      class TNEntry final : public ASN1_Object {
+         public:
+            enum Type { TN_ServiceProviderCode = 0, TN_TelephoneNumberRange, TN_TelephoneNumber };
+
+            struct TelephoneNumberRange {
+                  ASN1_String start;  //TelephoneNumber (IA5String)
+                  size_t count;       //2..MAX
+            };
+
+            TNEntry() = default;
+            virtual ~TNEntry() = default;
+
+            void encode_into(DER_Encoder&) const override;
+            void decode_from(class BER_Decoder& from) override;
+
+            Type get_type() const { return type; }
+
+            const std::string& getServiceProviderCode() const { return spc.value(); }
+
+            const std::vector<TelephoneNumberRange>& getTelephoneNumberRange() const { return range; }
+
+            const std::string& getTelephoneNumber() const { return one.value(); }
+
+         private:
+            Type type;
+            ASN1_String spc;                          //ServiceProviderCode
+            std::vector<TelephoneNumberRange> range;  //TelephoneNumberRange
+            ASN1_String one;                          //TelephoneNumber
```

Comment:
Members in Botan are always prefixed with `m_`:

```suggestion
            Type m_type;
            ASN1_String m_spc;                          //ServiceProviderCode
            std::vector<TelephoneNumberRange> m_range;  //TelephoneNumberRange
            ASN1_String m_one;                          //TelephoneNumber
```